### DISCUSSION TITLE
Accordion item limit before noticeable slowdown of render time?

### DISCUSSION THREAD

**Discussion by overpolish** [chakra-ui/chakra-ui#4554] [Q&A]
Hi, I am using an Accordion which is working great, in my application users are able to dynamically add new objects which in turn creates new accordion buttons and panels. Each panel has a single input field (for now), at around 200 items there is noticeable slowdown in expanding/collapsing an accordion panel.

Does anyone know the cause of this slow down? Using the input fields alone causes no noticeable slowdown past 200 with this system. Re-renders are isolated to only new components being added and thus the issue is purely a numbers thing.

**Comment by payapula** [comment]
Seems like a interesting issue. I have a similar setup with dynamic panels in accordion but haven't gone past 10 (at most). 

It would be easy to debug/track if you can create a reproduction in codesandbox and share here. Thank you.

**Reply by overpolish** [reply]
I have created the [codesandbox](https://codesandbox.io/s/beautiful-night-y4vs4). In the **App.js** you can vary the count to generate a different numbers of accordion items. I suggest to try 25 and then 250 to see the different in response times.

**Reply by payapula** [reply]
Thanks for the sandbox, quickly sent it through the profiler and found that all 250 panels are re-rendered because of `AccordianContext.Provider` changed (because of hooks, the classic case with useContext.) 

![image](https://user-images.githubusercontent.com/7134153/129209590-d84b4d9e-b6a3-49a1-b606-3d25603cc96a.png)

Since it is not anything that we the users can control, I believe it has to be looked further by chakra team.

**Comment by natromankevich** [comment]
Having the same issue using chakra-ui v1.8.8.

**Comment by SiyrisSoul** [comment]
Having the same issue with `chakra-ui/react v2.4.1`. Very noticeable on mobile and when using accordions in a modal.

**Comment by florind-ens** [comment]
The same issue today. I wonder if anyone found some workarounds.

**Comment by noyon-bks** [comment]
Having same issue in `chakra-ui/react v2.0.17`

**Comment by tahirTechSea** [comment]
having same issues still today

**Comment by sneuvonen** [comment]
I was experiencing similar issues with version 3 but lazyMount-prop solved them.
